cmd/utils.go

```diff
@@ -95,9 +97,65 @@ func splitStr(path, sep string, n int) []string {
 	return splits
 }
 
+// parse url usually obtained from env.
+func parseEnvURL(envURL string) (*url.URL, string, string, *probe.Error) {
+	u, e := url.Parse(envURL)
+	if e != nil {
+		return nil, "", "", probe.NewError(e).Trace(envURL)
+	}
+
+	var accessKey, secretKey string
+	if u.User != nil { // Check if username:password is provided in URL.
+		accessKey = u.User.Username()
+		secretKey, _ = u.User.Password()
+	} // Empty access and secret keys are okay, we just perform anonymous requests.
```

Comment:
You could have this comment above the `if` block for better understanding

Reply:
Let me add` // else`

Reply:
A comment before `if` would help reading comments before the code.   If you see a code itself is clear enough, you can remove comments.